perceval/runtime/_token_management.py

```diff
@@ -71,13 +77,15 @@ def _from_file(self) -> str | None:
                 get_logger().warn("Cannot read token persistent file", channel.user)
         return token
 
+    @deprecated(version="0.13.0", reason=f"Use instead RemoteConfig class method `get_token`")
     def get_token(self) -> str | None:
         """Search for a token to provide
 
         :return: A token, or None if no token was found
         """
         return TokenProvider._CACHED_TOKEN or self._from_environment_variable() or self._from_file()
 
+    @deprecated(version="0.13.0", reason=f"Use instead RemoteConfig class methods `set_token` then `save`")
```

Comment:
I think you can safely remove all the private methods (starting with _ ). For the other methods, you should make them continue to work, so you should call the methods from `RemoteConfig` internally, as it's the one that is called in the `RemoteProcessor`. The `TokenProvider` should no longer do anything on its own